test/fixtures/goals.yml

```diff
@@ -1,2 +1,3 @@
 goal_1:
   title: Goal 1
+  public_uid: <%= SecureRandom.random_number(1_000_000) %> # generate public_uid
```

Comment:
@laura-murria-nesta the reason the tests were failing was because `Goal.find_puid(nil)` was finding a Goal record with a `public_uid` of `nil` and setting the `@targetable` instance variable in the indicators controller to a Goal and not the expected Project. This was likely due to the fact that active record callbacks aren't called with fixtures. The workaround was to generate the public_uid in the fixture itself (and also in projects.yml). This also meant that I was able to remove the `._run_create_callbacks` calls throughout various tests.